### DISCUSSION TITLE
convolution layer when cudnn_off = False will cause random result

### DISCUSSION THREAD

**Discussion by xiaodao2049** [apache/incubator-mxnet#14941] [Q&A] [answered]
Hi,every one,
I'm training a net in which some layers are convolution layer, and I set a random seed, but when the   
cudnn_off = False the result will be random, and if True, the result is reproduceable.

[a comment by an automated account is not shown]

**Comment by vdantu** [comment, marked as answer]
@xiaodao2049 : I think the reason is that there are multiple implementations of convolution when using cundd  and cudnn chooses the best algorithm based on your available system resources at runtime. You could try using "MXNET_ENFORCE_DETERMINISM" (Refer [this PR](https://github.com/apache/incubator-mxnet/pull/12992) for full context) and see if this helps. If not you could turn off autotune by setting "MXNET_CUDNN_AUTOTUNE_DEFAULT" to 0 and see if that helps. 

@apeforest , please correct me if any part of my answer is misrepresented :) 

@mxnet-label-bot add [question, cudnn]

**Comment by vdantu** [comment]
@mxnet-label-bot update [question]

**Comment by xiaodao2049** [comment]
@vdantu thanks for your reply. I have tried both solution you mentioned,MXNET_ENFORCE_DETERMINISM doesn't work, maybe my mx version is too old. Setting "MXNET_CUDNN_AUTOTUNE_DEFAULT" to 0 thrown an std::bad_alloc error. I will upgrade my mxnet and try the first solution again.
